aea/protocols/generator.py

```diff
@@ -65,16 +66,146 @@ def __init__(
         self.protocol_specification = protocol_specification
         self.output_folder_path = os.path.join(output_path, protocol_specification.name)
 
-    def _extract_all_contents(self) -> Dict[str, Dict[str, str]]:
-        all_contents = {}  # type: Dict[str, Dict[str, str]]
+        self._imports = {
+            "Set": True,
+            "Tuple": True,
+            "cast": True,
+            "Dict": False,
+            "Union": False,
+            "Optional": False,
+            "FrozenSet": False,
+        }
+
+        self._speech_acts = dict()  # type: Dict[str, Dict[str, str]]
+        self._all_performatives = set()  # type: Set[str]
+        self._all_unique_contents = dict()  # type: Dict[str, str]
+        self._all_custom_types = set()  # type: Set[str]
+
+        self._setup()
+
+    def _setup(self) -> None:
+        """
+        Extract all relevant data structures from the specification.
+
+        :return: Dict[performatives, Dict[content names, content types]]
+        """
         for (
             performative,
             speech_act_content_config,
         ) in self.protocol_specification.speech_acts.read_all():
-            all_contents[performative] = {}
+            self._all_performatives.add(performative)
+            self._speech_acts[performative] = {}
             for content_name, content_type in speech_act_content_config.args.items():
-                all_contents[performative][content_name] = content_type
-        return all_contents
+                custom_types = set(re.findall(CUSTOM_TYPE_PATTERN, content_type))
+                for ct in custom_types:
```

Comment:
please write out variable names, so `ct` -> `custom_type` in this case

Reply:
Done